### PR TITLE
[Enhancement] Use the MUI date picker for date inputs

### PR DESCRIPTION
**Pull request by roseeichelmann**
## Associated issues
Fixes cityofaustin/atd-data-tech/issues/10977

## Testing
**URL to test:** 
https://deploy-preview-925--atd-moped-main.netlify.app/moped/ 

**Steps to test:**
1. Go to the timelime tab of a project
2. Add a new phase or milestone, play around with adding new start and end dates, clearing the dates, editing them, etc.
3. Admire the new datepicker modal

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/DateFieldEditComponent.js [hunk 1]
```diff
@@ -9,19 +10,22 @@ import { handleKeyEvent } from "../../../utils/materialTableHelpers";
...
+      <DatePicker
+        clearable={true}
+        emptyLabel="mm/dd/yyyy"
+        format="MM/dd/yyyy"
+        value={props.value ? parseISO(props.value) : null}
+        onChange={(e) => {
+          const newDate = e ? format(e, "yyyy-MM-dd") : null;
+          props.onChange(newDate);
+        }}
+        InputProps={{ style: { minWidth: "100px" } }}
```
Comment (roseeichelmann) [inline moped-editor/src/views/projects/projectView/DateFieldEditComponent.js:25]: the input value was cutting off at certain zooms, this mitigates that

File: moped-editor/package.json [hunk 2]
```diff
@@ -41,13 +41,15 @@
   },
   "dependencies": {
     "@apollo/client": "^3.6.9",
+    "@date-io/date-fns": "^1.3.13",
```
Comment (johnclary) [inline moped-editor/package.json:44]: just pointing out that MUI requires this `date-fns` wrapper — despite our project already having the `date-fns` package included

**Review by johnclary — APPROVED** [review]
looks great; works great. thank you @roseeichelmann!! 🚢 🚢 🚢 

please hold off on merging until another dev reviews the code.

**Comment by chiaberry** [comment]
I've admired the new date picker, it looks good! But I think I found a bug. 
I am trying to add a new phase: https://deploy-preview-925--atd-moped-main.netlify.app/moped/projects/1907?tab=timeline

I filled out a new phase with phase, subphase, start date and end date, but did not toggle the phase as current. and i get this error
Screen Shot 2023-01-04 at 12 21 12 PM

**Comment by johnclary** [comment]
the issue @chiaberry ran into is b/c the staging db has diverged from this PR via https://github.com/cityofaustin/atd-moped/pull/906. i believe merging main into here would fix this issue?

**Comment by roseeichelmann** [comment]
@chiaberry @johnclary i just merged main into this branch, lmk if that clears up the issue!

**Comment by johnclary** [comment]
thanks @roseeichelmann - that worked for me!

File: moped-editor/src/views/projects/projectView/ProjectPhases.js [hunk 3]
```diff
@@ -173,9 +173,7 @@ const ProjectPhases = ({
         rowData.phase_start
           ? format(parseISO(rowData.phase_start), "MM/dd/yyyy")
           : undefined,
-      editComponent: (props) => (
-        <DateFieldEditComponent {...props} name="phase_start" label="Start" />
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectPhases.js:177]: Really like that these redundant labels are removed 👏

File: moped-editor/src/views/projects/projectView/DateFieldEditComponent.js [hunk 4]
```diff
@@ -9,19 +10,22 @@ import { handleKeyEvent } from "../../../utils/materialTableHelpers";
...
+    <MuiPickersUtilsProvider utils={DateFnsUtils}>
+      <DatePicker
+        clearable={true}
+        emptyLabel="mm/dd/yyyy"
+        format="MM/dd/yyyy"
+        value={props.value ? parseISO(props.value) : null}
+        onChange={(e) => {
+          const newDate = e ? format(e, "yyyy-MM-dd") : null;
+          props.onChange(newDate);
+        }}
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/DateFieldEditComponent.js:24]: Not a super strong opinion but this onChange handler function could be defined in body of this component

**Review by mddilley — APPROVED** [review]
Step 3 complete ✅ ✨ The other steps worked great too. 🚀

**Review by chiaberry — APPROVED** [review]

File: moped-editor/src/views/projects/projectView/DateFieldEditComponent.js (hunk 4 above)
Comment (roseeichelmann) [reply moped-editor/src/views/projects/projectView/DateFieldEditComponent.js]: I just added this, thanks for the comment I totally agree!

**Pull request merged**